src/App.tsx

```diff
@@ -14,11 +15,130 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const goodsList = [...goodsFromServer];
+  const [initialButtonVisible, start] = useState(true);
+  const [isReversed, reverse] = useState(false);
+  const [isSortedByLength, sortByLength] = useState(false);
+  const [isSortedByName, sortByName] = useState(false);
+  const [isReset, reset] = useState(false);
+  const [goodsLength, setGoodsLength] = useState('1');
+  const visibleGoods = goodsList.filter(good => good.length >= +goodsLength);
```

Comment:
Try to use `Number` constructor it's more obvious
```suggestion
  const visibleGoods = goodsList.filter(good => good.length >= Number(goodsLength));
```